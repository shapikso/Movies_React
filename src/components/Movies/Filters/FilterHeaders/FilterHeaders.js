import React from 'react';
import Select from "../Selector/Select";
import {movieLanguages, movieStatus} from "../../../../constants/selectors";
import Input from "../../../common/input/InputField";
import PropTypes from "prop-types";

const FilterHeaders = ({setFilter, title}) => {
    const setStatus = (status) => setFilter('status', status);
    const setLanguage = (language) => setFilter('language', language);
    const setSearchTitle = (title) => setFilter('title', title);

    return (
        <div className="form-filters__header">
            <Input value={title}
                placeholder="Search..."
                onChange={setSearchTitle}
                className="movie-title"
            />
            <Select setSelector={setStatus} options={movieStatus}/>
            <Select setSelector={setLanguage} options={movieLanguages}/>
        </div>
    );
};

FilterHeaders.propTypes = {
    setFilter: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired
};

export default FilterHeaders;