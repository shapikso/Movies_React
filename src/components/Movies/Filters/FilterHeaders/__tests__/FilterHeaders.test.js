import React from 'react';
import {shallow, mount} from 'enzyme';
import FilterHeaders from "../FilterHeaders";

describe('FilterHeaders', ()=> {
    const props = {
        title: 'Search title',
        language: 'jp',
        status: 'Released',
        setFilter: jest.fn(),
    };
    it('should render correctly', () => {
        const component = shallow(<FilterHeaders {...props} />);
        expect(component).toMatchSnapshot();
    });
    it('should checked render title', () => {
        const component = mount(<FilterHeaders {...props} />);
        expect(component.find('InputField').getElement().props.value).toEqual(props.title);
    });
    it('should changed title', () => {
        const component = mount(<FilterHeaders {...props} />);
        component.find('input').getElement().props.onChange({target: { value : props.title}});
        expect(props.setFilter).toHaveBeenCalledWith('title',props.title);
    });
    it('should changed language', () => {
        const component = mount(<FilterHeaders {...props} />);
        component.find('select').find('select').at(1).getElement().props.onChange({target: { value :  props.language}});
        expect(props.setFilter).toHaveBeenCalledWith('language', props.language);
    });
    it('should changed status', () => {
        const component = mount(<FilterHeaders {...props} />);
        component.find('select').find('select').at(0).simulate('change', {target: { value : props.status}});
        expect(props.setFilter).toHaveBeenCalledWith('status', props.status);
    });

});