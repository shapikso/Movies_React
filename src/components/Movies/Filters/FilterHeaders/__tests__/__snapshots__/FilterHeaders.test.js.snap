// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterHeaders should render correctly 1`] = `ShallowWrapper {}`;
